src/build_vehicle_gui.cpp

```diff
@@ -1785,13 +1861,45 @@
 	{
 		this->vscroll->SetCapacityFromWidget(this, WID_BV_LIST);
 	}
+
+	void OnEditboxChanged(int wid) override
+	{
+		if (wid == WID_BV_FILTER) {
+			this->string_filter.SetFilterTerm(this->vehicle_editbox.text.buf);
+			this->InvalidateData();
+		}
+	}
+
+	EventState OnHotkey(int hotkey) override
+	{
+		switch (hotkey) {
+			case BVHK_FOCUS_FILTER_BOX:
+				this->SetFocusedWidget(WID_BV_FILTER);
+				SetFocusedWindow(this); // The user has asked to give focus to the text box, so make sure this window is focused.
+				break;
+
+			default:
+				return ES_NOT_HANDLED;
+		}
```

Comment:
Should this be fixed or dismissed? Hotkey handling, like GUI stuff, tends to use switches so it's a question of consistency.

Reply:
I think dismissing them is counter productive, as then you would probably dismiss all of them and any new cases get this alert again.
So, there is a choice to be made: 
* do we want to keep this coding style, then explicitly disable this specific check from the CodeQL configuration (.github/codeql/codeql-config.yml, where two are already disabled).
* do we want to comply with this alert, then make a (separate) PR that changes all these cases from a `switch` to an `if`.

I would have a slight favour for the first solution just because this makes extending these kinds of callbacks easier in the future. But I can also live with changing them to a simple `if`

On the other hand, the `OnEditboxChanged` above it uses the `if` paradigm.
Conceptually, I would replace the break with `return ES_HANDLED` as you're done handling the event at that point.

Reply:
I'll make a separate PR proposing to change the CodeQL configuration. That seems like a better place to have this discussion.

`OnEditboxChanged` seems to use `if` globally when there's only one case, except for a few uses where it doesn't even check the widget. The only use of `switch` is the network game list, which has two editboxes.

Reply:
Resolved by #10537.